### ISSUE TITLE
Can't export view as table with 23.1 and pg11

### ISSUE THREAD

**Issue by vicvtor** [darold/ora2pg#1491]
Hello!
I'm trying to start migrating data from Oracle 11.2 to Pg 11, while scanning data on the Oracle side breaks, without any errors. I will assume that there are problems with the environment, but it is not clear which way to look. Perhaps you can give me some advice. Launch log attached below. Thanks in advance!
```
root@test:~# cd /opt/ora2pg/
root@test:/opt/ora2pg/# ora2pg -c ./ora2pg.conf --basedir ./
[2022-09-05 14:49:18] Ora2Pg version: 23.1
[2022-09-05 14:49:18] Export type: COPY
[2022-09-05 14:49:18] Geometry export type: INTERNAL
[2022-09-05 14:49:18] ORACLE_HOME = /usr/lib/oracle/11.2/client64
[2022-09-05 14:49:18] NLS_LANG = AMERICAN_AMERICA.AL32UTF8
[2022-09-05 14:49:18] NLS_NCHAR = AL32UTF8
[2022-09-05 14:49:18] Trying to connect to database: dbi:Oracle:host=SERVER;sid=DBSID;port=1521
[2022-09-05 14:49:18] Isolation level: SET TRANSACTION ISOLATION LEVEL SERIALIZABLE
[2022-09-05 14:49:18] Retrieving table information...
[2022-09-05 14:49:18] Collecting 1 tables in ALL_OBJECTS took:  0 wallclock secs ( 0.00 usr +  0.00 sys =  0.00 CPU)
[2022-09-05 14:49:19] Collecting 0 tables information in ALL_TABLES took:  1 wallclock secs ( 0.00 usr +  0.00 sys =  0.00 CPU)
[2022-09-05 14:49:20] Collecting 2827 tables in ALL_OBJECTS took:  1 wallclock secs ( 0.01 usr +  0.00 sys =  0.01 CPU)
[2022-09-05 14:51:26] Scanning view TEST_TABLE to export as table...
[2022-09-05 14:51:27] Collecting column information for table TEST_TABLE...
[2022-09-05 14:51:27] DEBUG,
SELECT A.COLUMN_NAME, A.DATA_TYPE, A.DATA_LENGTH, A.NULLABLE, A.DATA_DEFAULT,
    A.DATA_PRECISION, A.DATA_SCALE, A.CHAR_LENGTH, A.TABLE_NAME, A.OWNER
FROM ALL_TAB_COLUMNS A
WHERE 1=1 AND A.OWNER='DBSID'  AND (REGEXP_LIKE(UPPER(A.TABLE_NAME), ?))
ORDER BY A.COLUMN_ID
[2022-09-05 14:51:27] Collecting 14 columns in ALL_INDEXES took:  0 wallclock secs ( 0.00 usr +  0.00 sys =  0.00 CPU)
[2022-09-05 14:51:27] MYDEBUG: 2
Aborting export...
```

**Comment by darold** [comment]
Try to set TRANSACTION to readonly in ora2pg.conf

**Comment by vicvtor** [comment]
Unfortunately, it didn't help. Can any other information about the environment help?

`root@test:/opt/ora2pg# ora2pg -c ./ora2pg.conf --basedir ./
[2022-09-06 04:59:14] Ora2Pg version: 23.1
[2022-09-06 04:59:14] Export type: COPY
[2022-09-06 04:59:14] Geometry export type: INTERNAL
[2022-09-06 04:59:14] ORACLE_HOME = /usr/lib/oracle/11.2/client64
[2022-09-06 04:59:14] NLS_LANG = AMERICAN_AMERICA.AL32UTF8
[2022-09-06 04:59:14] NLS_NCHAR = AL32UTF8
[2022-09-06 04:59:14] Trying to connect to database: dbi:Oracle:host=SERVER;sid=DBSID;port=1521
[2022-09-06 04:59:15] Isolation level: SET TRANSACTION READ ONLY
[2022-09-06 04:59:15] Retrieving table information...
[2022-09-06 04:59:15] Collecting 1 tables in ALL_OBJECTS took:  0 wallclock secs ( 0.00 usr +  0.00 sys =  0.00 CPU)
[2022-09-06 04:59:17] Collecting 0 tables information in ALL_TABLES took:  2 wallclock secs ( 0.00 usr +  0.00 sys =  0.00 CPU)
[2022-09-06 04:59:19] Collecting 2827 tables in ALL_OBJECTS took:  2 wallclock secs ( 0.01 usr +  0.00 sys =  0.01 CPU)
[2022-09-06 05:01:35] Scanning view TEST_TABLE to export as table...
[2022-09-06 05:01:35] Collecting column information for table TEST_TABLE...
[2022-09-06 05:01:35] DEBUG,
SELECT A.COLUMN_NAME, A.DATA_TYPE, A.DATA_LENGTH, A.NULLABLE, A.DATA_DEFAULT,
    A.DATA_PRECISION, A.DATA_SCALE, A.CHAR_LENGTH, A.TABLE_NAME, A.OWNER
FROM ALL_TAB_COLUMNS A
WHERE 1=1 AND A.OWNER='DBSID'  AND (REGEXP_LIKE(UPPER(A.TABLE_NAME), ?))
ORDER BY A.COLUMN_ID
[2022-09-06 05:01:35] Collecting 14 columns in ALL_INDEXES took:  0 wallclock secs ( 0.00 usr +  0.00 sys =  0.00 CPU)
[2022-09-06 05:01:35] MYDEBUG: 2
Aborting export...`

**Comment by darold** [comment]
You must first check dmesg to see if there is a OOMKiller action.

Otherwise, you might try to use latest development code you could be experiencing a bug fixed by commit 2c8ca418.

**Comment by darold** [comment]
Closing, this should be fixed with latest development code.
